tests/generators/test_hexagonal.py

```diff
@@ -0,0 +1,57 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+import retworkx
+
+
+class TestHexagonalLatticeGraph(unittest.TestCase):
+
+    def test_directed_hexagonal_graph_2_2(self):
+        graph = retworkx.generators.directed_hexagonal_lattice_graph(2, 2)
+        self.assertEqual(len(graph), 16)
+        self.assertEqual(len(graph.edges()), 19)
+
+    def test_directed_hexagonal_graph_3_2(self):
+        graph = retworkx.generators.directed_hexagonal_lattice_graph(3, 2)
+        self.assertEqual(len(graph), 22)
+        self.assertEqual(len(graph.edges()), 27)
+
+    def test_directed_hexagonal_graph_2_4(self):
+        graph = retworkx.generators.directed_hexagonal_lattice_graph(2, 4)
+        self.assertEqual(len(graph), 28)
+        self.assertEqual(len(graph.edges()), 35)
+
+    def test_directed_hexagonal_graph_0_0(self):
+        graph = retworkx.generators.directed_hexagonal_lattice_graph(0, 0)
+        self.assertEqual(len(graph), 0)
+        self.assertEqual(len(graph.edges()), 0)
+
+    def test_hexagonal_graph_2_2(self):
+        graph = retworkx.generators.hexagonal_lattice_graph(2, 2)
+        self.assertEqual(len(graph), 16)
+        self.assertEqual(len(graph.edges()), 19)
```

Comment:
It'd be nice to verify that this is actually a hexagonal lattice so we know moving forward that if we make changes that we're not breaking this. Right now all we know is that there are 16 nodes and 19 edges in the graph, not how they're connected.

Reply:
Sure, I'll add that.